contracts/contracts/pausing/Pauser.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+import { Governable } from "../governance/Governable.sol";
+import { Initializable } from "../utils/Initializable.sol";
+import { Pausable } from "./Pausable.sol";
+
+contract Pauser is Initializable, Governable {
+    event ExpiryDurationChanged(uint256 _newExpiry);
+    event PausableChanged(address _newPausable);
+    event StrategistChanged(address _newStrategist);
+    event TempPaused(uint256 _expiry);
+    event Paused();
+    event Unpaused();
+    event Whitelisted(address _account);
+    event DeWhitelisted(address _account);
+
+    uint8 public constant UNPAUSED = 0;
+    uint8 public constant TEMP_PAUSE = 1;
+    uint8 public constant PAUSED = 2;
+
+    /**
+     * @dev The current state of the contract.
+     */
+    uint8 public pauseState;
+
+    /**
+     * @dev The duration of a temp-pause in seconds.
+     */
+    uint256 public expiryDuration;
+
+    /**
+     * @dev The expiry timestamp of the current temp-pause.
+     * If unpaused or confirmed, the expiry timestamp should be zero (0).
+     */
+    uint256 public currentExpiry;
+
+    /**
+     * @dev The address of the managed contract.
+     */
+    address public pausable;
+
+    /**
+     * @dev Address of the strategist
+     */
+    address public strategistAddr = address(0);
+
+    /**
+     * @dev whitelist of addresses allowed to interact with the functionality exposed by this contract.
+     */
+    mapping(address => bool) public whitelist;
```

Comment:
Rather than writing our own code to add and remove whitelisted pausers, perhaps we should use the OpenZeppelin access control library.

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/AccessControl.sol

Reply:
Good idea. Makes it easier to scale in case the requirements change and we need more roles.